src/Symfony/Component/Notifier/Bridge/SmsBiuras/SmsBiurasTransport.php

```diff
@@ -0,0 +1,103 @@
+    private $uid;
+    private $apiKey;
+    private $from;
+    private $test;
+
+    private const ERROR_CODES = [
+        1 => 'The message was processed and sent to the mobile operator. But delivery confirmations have not yet been returned.',
+        2 => 'SMS not delivered.',
+        3 => 'The SMS message was successfully delivered to the recipient.',
+        4 => 'The message was sent and expired because it could not be delivered to the recipient during its validity period (48 hours according to our default platform).',
+        5 => 'The message was received but the operator returned "Rejected" as the final status.',
+        6 => 'Missing parameters, check that you are using all required parameters.',
+        7 => ' Wrong apikey or uid.',
+        8 => 'Sender ID - "from". Must be approved by an administrator.',
+        9 => 'Balance insufficient, please top up the account.',
+        10 => 'Bad date format for schedule parameter. Ex: urlencode("2021-03-11 12:00").',
+    ];
+
+    public function __construct(string $uid, string $apiKey, string $from, int $test, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->uid = $uid;
+        $this->apiKey = $apiKey;
+        $this->from = $from;
+        $this->test = $test;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('smsbiuras://%s?from=%s&test=%s', $this->getEndpoint(), $this->from, $this->test);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $endpoint = 'https://'.$this->getEndpoint().'/api?'.http_build_query([
+                'uid' => $this->uid,
+                'apikey' => $this->apiKey,
+                'message' => $message->getSubject(),
+                'from' => $this->from,
+                'to' => $message->getPhone(),
+                'test' => $this->test,
+        ]);
+
+        $response = $this->client->request('GET', $endpoint);
+
+        if (Response::HTTP_OK !== $response->getStatusCode()) {
+            throw new TransportException('Unable to send the SMS: Service Unavailable.', $response);
+        }
+
+        $matches = [];
+        if (preg_match('/^ERROR: (\d+)$/', $response->getContent(), $matches)) {
+            throw new TransportException('Unable to send the SMS: '.self::ERROR_CODES[$matches[1]], $response);
+        }
+
+        $matches = [];
+        if (preg_match('/^OK: (\d+)$/', $response->getContent(), $matches)) {
+            $sentMessage = new SentMessage($message, (string) $this);
+            $sentMessage->setMessageId($matches[1]);
+        }
+
+        return $sentMessage;
```

Comment:
This can be undefined

Reply:
In theory yes. In practice should be SMS ID :) Fixed

Reply:
The variable can still be undefined